### DISCUSSION TITLE
[Q/A][Java SDK (Component)] dubbo 3.210 provider service set retries = 0 Not working

### DISCUSSION THREAD

**Discussion by stupidboytt** [apache/dubbo#14192] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

provider service set
@DubboService(methods = {
		@Method(name = "insert", retries = 0)
})

yml setting
dubbo:
  application:
    name: provider
  provider:
    timeout: 10000

Set the retry count to 0, but after the interface timeout, it was found that 3 pieces of data were inserted into the database
why？

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Can you provide a reproducable demo?

**Reply by stupidboytt** [reply]
@DubboReference(methods = {
            @Method(name = "insert", retries = -1)
    })
    private DubboService dubboService;

![image](https://github.com/apache/dubbo/assets/52768038/92482e4a-dd02-424a-873b-359d2afa238c)

The retries I set=-1, but in the FailoverClusterInvoker.calculateInvokeTimes API, the retries I obtained=null, then the len is three, finally call API 3 times，
why set retries = -1 ，finally obtained null

**Reply by AlbumenJ** [reply]
Have you tried with 3.2.12?

**Reply by stupidboytt** [reply]
dubbo version is 3.2.10

**Reply by AlbumenJ** [reply]
In my memory, this has been fix in the latest version. So please check it :)
